src/main/java/dev/jbang/Util.java

```diff
@@ -669,4 +705,45 @@ private static boolean createHardLink(Path src, Path target) {
 		return true;
 	}
 
+	public static Path getUrlCache(String fileURL) {
+		String urlHash = getStableID(fileURL);
+		return Settings.getCacheDir(Settings.CacheClass.urls).resolve(urlHash);
+	}
+
+	public static Path getPathToFile(String scriptURL, File baseDir) {
+		String fileName = scriptURL.substring(scriptURL.lastIndexOf("/") + 1);
+		return baseDir.toPath().resolve(fileName);
+	}
+
+	public static boolean hasMainMethod(String content) {
+		// (public[ ]{1,}static[ ]{1,}void[ ]{1,}main[ ]{0,}\()|(static[ ]{1,}public[
+		// ]{1,}void[ ]{1,}main[ ]{0,}\()
+		// TODO create regular expression
+		return content.contains("public static void main(");
+	}
+
+	public static boolean isGistURL(String scriptURL) {
+		return scriptURL.startsWith("https://gist.github.com/");
+	}
+
+	public static List<String> collectSources(List<String> lines) {
```

Comment:
this and the other collectSources() feels like should be on Script class?

Reply:
> this and the other collectSources() feels like should be on Script class?

I moved them to the Script class.